sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/lang/SqlDelightQueriesFile.kt

```diff
@@ -107,14 +111,32 @@ class SqlDelightQueriesFile(
    * A collection of all the adapters needed for arguments or result columns in this query.
    */
   internal val requiredAdapters by lazy {
-    val argumentAdapters = PsiTreeUtil.findChildrenOfType(this, SqlBindExpr::class.java)
-      .mapNotNull { typeResolver.argumentType(it).parentAdapter() }
+    val binders = PsiTreeUtil.findChildrenOfType(this, SqlBindExpr::class.java)
+    val argumentAdapters = binders.singleQuestionMark() ?: binders.mapNotNull {
+      typeResolver.argumentType(it).parentAdapter()
```

Comment:
you'll want to have your logic live in this function and have it be invoked as a `flatMap` so that you can expand a single `SqlBindExpr` to multiple `PropertySpec`

Reply:
and the logic would be

```
.flatMap {
  it.parentOfType<SqlInsertStmt>()?.let {
    if (it.acceptsTableInterface()) return@flatMap listOf(table.adapterProperty())
  }
  return listOf(typeResolver.argumentType(it).parentAdapter())
}.filterNotNull()
```

Reply:
Don't you need `if (it.acceptsTableInterface() && it.needsAdapter())` to prevent adding unused adapter property?

Reply:
ah right, because the table has only one adapter not multiple. Yes